python/pyspark/sql/context.py

```diff
@@ -343,28 +344,29 @@ def createDataFrame(
 
     @overload
     def createDataFrame(
-        self, data: "PandasDataFrameLike", samplingRatio: Optional[float] = ...
+        self, data: Union["PandasDataFrameLike", "pa.Table"], samplingRatio: Optional[float] = ...
```

Comment:
can we include `pa.Table` in `PandasDataFrameLike`?

Reply:
I don't think so. `PandasDataFrameLike` is just an alias for `pd.DataFrame`.

In `pyspark.sql.pandas._typing`, `DataFrameLike` is defined like this:
```py
from pandas.core.frame import DataFrame as PandasDataFrame
DataFrameLike = PandasDataFrame
```

Then in other parts of PySpark, `PandasDataFrameLike` is defined like this:
```py
from pyspark.sql.pandas._typing import DataFrameLike as PandasDataFrameLike
```

If we define `PandasDataFrameLike` as a Union that includes `pa.Table`, that will cause other problems. For example, then we can't use it as the return type of `toPandas()`.